apx/common/src/apx_fileManager.c

```diff
@@ -405,7 +403,7 @@ const char *apx_fileManager_modeString(apx_fileManager_t *self)
 
 void apx_fileManager_onConnected(apx_fileManager_t *self)
 {
-   if (self != 0)
+   if (self != 0 && (self->workerThreadValid == true))

```

Comment:
1. You must enclose first expression in parenthesis: `(self != 0)`
2. Since workerThreadValid is declared as type bool it is regarded as "essentially boolean" by MISRA standard.
Therefore, it is not necessary to compare against explicit value true.

Final Form:
`if ( (self != 0) && (self->workerThreadValid))`